src/catalog/src/durable/transaction.rs

```diff
@@ -785,6 +822,48 @@ impl<'a> Transaction<'a> {
         }
     }
 
+    /// Removes all schemas in `schemas` from the transaction.
+    ///
+    /// Returns an error if any id in `schemas` is not found.
+    ///
+    /// NOTE: On error, there still may be some schemas removed from the transaction. It
+    /// is up to the caller to either abort the transaction or commit.
+    pub fn remove_schemas(
+        &mut self,
+        schemas: &BTreeMap<SchemaId, ResolvedDatabaseSpecifier>,
+    ) -> Result<(), CatalogError> {
+        let to_remove = schemas
+            .iter()
+            .map(|(schema_id, _)| (SchemaKey { id: *schema_id }, None))
+            .collect();
+        let mut prev = self.schemas.set_many(to_remove, self.op_id)?;
+        prev.retain(|_k, v| v.is_none());
+
+        if !prev.is_empty() {
+            let err = prev
+                .keys()
+                .map(|k| {
+                    let db_spec = schemas.get(&k.id).expect("should_exist");
+                    let db_name = match db_spec {
+                        ResolvedDatabaseSpecifier::Id(id) => format!("{id}."),
+                        ResolvedDatabaseSpecifier::Ambient => "".to_string(),
+                    };
+                    format!("{}.{}", db_name, k.id)
+                })
+                .join(", ");
+
+            return Err(SqlCatalogError::UnknownSchema(err).into());
+        }
```

Comment:
Not covered by any test according to the [coverage report](https://buildkite.com/materialize/coverage/builds/414). Could you add a (unit) test?

Reply:
Added!